examples/chatgpt/rag-quickstart/azure/Azure_AI_Search_with_Azure_Functions_and_GPT_Actions_in_ChatGPT.ipynb

```diff
@@ -0,0 +1,1330 @@
+                "print(\"Search Service Endpoint:\" + search_service_endpoint)\n"
+            ]
+        },
+        {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "#### Get the API Key\n",
+                "This grabs the Search Service API Key, which we'll use to initiate the index creation, and later to do the search"
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "\n",
+                "# Initialize the SearchManagementClient with the provided credentials and subscription ID\n",
+                "search_management_client = SearchManagementClient(\n",
+                "    credential=credential,\n",
+                "    subscription_id=subscription_id,\n",
+                ")\n",
+                "\n",
+                "# Retrieve the admin keys for the search service\n",
+                "response = search_management_client.admin_keys.get(\n",
+                "    resource_group_name=resource_group,\n",
+                "    search_service_name=search_service_name,\n",
+                ")\n",
+                "\n",
+                "# Extract the primary API key from the response\n",
+                "search_service_api_key = response.primary_key\n",
+                "\n",
+                "# Append the search_service_api_key to the .env file\n",
+                "with open('.env', 'a') as env_file:\n",
+                "    env_file.write('\\n')  # Add a newline for separation\n",
+                "    env_file.write(f'SEARCH_SERVICE_API_KEY=\"{search_service_api_key}\"\\n')  # Write the API key to the file\n",
+                "\n"
+            ]
+        },
+        {
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "### Create the initial index\n",
+                "Choose a name for the index below, or use the default provided."
+            ]
+        },
+        {
+            "cell_type": "code",
+            "execution_count": null,
+            "metadata": {},
+            "outputs": [],
+            "source": [
+                "\n",
+                "index_name = \"azure-ai-search-openai-cookbook-demo\"\n",
+                "# index_name = \"<insert_name_for_index>\"\n",
+                "\n",
+                "# Initialize the SearchClient with the selected authentication method\n",
+                "search_client = SearchClient(\n",
+                "    endpoint=search_service_endpoint, index_name=index_name, credential=AzureKeyCredential(search_service_api_key)\n",
+                ")"
+            ]
+        },
+        {
+            "attachments": {},
+            "cell_type": "markdown",
+            "metadata": {},
+            "source": [
+                "## Load data\n",
```

Comment:
Load Data is a long section as well; recommend trying to make everything a bit more concise and adding another roadmap at the beginning to orient the reader